### ISSUE TITLE
BitbucketStatusPush getting 200 on success instead of 201

### ISSUE THREAD

**Issue by dunderpate** [buildbot/buildbot#5284]
The documentation for the BitBucket status API claims that the successful response code is 201, however we are getting back 200 on a successful status update. I don't really think this is a problem with BuildBot, but perhaps an issue with the documentation on BitBucket's side. I've opened an issue for this and hope to hear back from them sometime soon.

This issue is more to track the fact that users may experience false positive errors in their logs at the moment due to this problem.

The error you'll see in the log:

```
2020-05-05 18:18:26+0000 [-] 200: unable to upload Bitbucket status b'{…}'
```

The documentation that claims 201 is the success code:

* https://archive.is/tklGp

The issue raised on BitBucket's issue tracker:

* https://getsupport.atlassian.com/servicedesk/customer/portal/11/BBS-135812

The BitbucketStatusPush documentation in BitBucket:

* http://docs.buildbot.net/latest/manual/configuration/reporters.html#bitbucketstatuspush

Dunderpate

**Comment by dunderpate** [comment]
Update from Atlassian:

> Just confirming that we got your request. We're on it.

**Comment by dunderpate** [comment]
Update from Atlassian:

> Sandeep Khandelwal
> 05/May/20 11:41 PM
>
> Hi there,
>
> Hope you are doing good and thanks for reaching out to us. I will be happy to assist in this case.
> The status code 200 is returned when the "The request has succeeded." and 201 is returned when  "The request has been fulfilled and has resulted in one or more new resources being created."
>
> You can get more details about status 200 and 201 in the below link -
>
> https://httpstatuses.com/200
> https://httpstatuses.com/201
>
> The API endpoint that you shared (build status POST) does create a new build status against the specified commit. 201 return code in this situation is appropriate as it does create a resource.
>
> Do let me know in case of more questions, will be happy to assist in this case.
>
> Thanks,
>
> Sandeep

**Comment by p12tic** [comment]
I think we should just allow both 201 and 200 as they both indicate success. Maybe you could do a PR?

**Comment by dunderpate** [comment]
My reply to Atlassian:

> Hi Sandeep!
>
> Thanks for the quick reply!
>
> Right, I agree and am familiar with de-facto standard REST status codes. I guess I just assumed all return codes would be documented in the BitBucket Cloud REST API documentation. I guess only return codes that have a payload are documented then? In any case, that explains why it appears incomplete. It's not designed to be.
>
> Thanks Sandeep,
>
> Dunderpate

**Comment by dunderpate** [comment]
Hi @p12tic,

Exactly what I was thinking. Very easy to do. I'll supply one, probably tomorrow morning. Thanks for the reply @p12tic.

Dunderpate

**Comment by dunderpate** [comment]
I've got a draft PR @ https://github.com/buildbot/buildbot/pull/5288. Once the tests pass, I've got one more push to clean up the commit message and then I think it'll be ready for review.

**Comment by dunderpate** [comment]
I think this one's ready for review now whenever you guys have time. Have a great night!

**Comment by dunderpate** [comment]
Update from Atlassian:

> Hi dunderpate
>
> Glad to hear this case is solved .
>
> We do really appreciate customer feedback to enhance the product and I am happy that you did let us know the probable issue with the documentation.
>
> I’m closing this ticket from now, but in case you require any assistance in the future, do not hesitate to contact us. You will receive a survey shortly, and any feedback is very much appreciated to help us improve.
>
> Cheers, 
>
> Sandeep

**Comment by p12tic** [comment]
Fixed in #5288.
